lntest/itest/lnd_test.go

```diff
@@ -4671,6 +4671,37 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
 		lnd.DefaultBitcoinTimeLockDelta, maxHtlc, carol,
 	)
 
+	// Before we start sending payments, subscribe to htlc events for each
+	// node.
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	aliceEvents, err := net.Alice.RouterClient.SubscribeHtlcEvents(
+		ctxt, &routerrpc.SubscribeHtlcEventsRequest{},
+	)
+	if err != nil {
+		t.Fatalf("could not subscribe events: %v", err)
+	}
+
+	bobEvents, err := net.Bob.RouterClient.SubscribeHtlcEvents(
+		ctxt, &routerrpc.SubscribeHtlcEventsRequest{},
+	)
+	if err != nil {
+		t.Fatalf("could not subscribe events: %v", err)
+	}
+
+	carolEvents, err := carol.RouterClient.SubscribeHtlcEvents(
+		ctxt, &routerrpc.SubscribeHtlcEventsRequest{},
+	)
+	if err != nil {
+		t.Fatalf("could not subscribe events: %v", err)
+	}
+
+	daveEvents, err := dave.RouterClient.SubscribeHtlcEvents(
+		ctxt, &routerrpc.SubscribeHtlcEventsRequest{},
+	)
+	if err != nil {
+		t.Fatalf("could not subscribe events: %v", err)
+	}
```

Comment:
Maybe it is possible to start with a `map[node]expectation`. Then subscribe in a loop to all map keys (the nodes) and later on assert on the events also in a loop. (brainstorming to see if the event assertions can be structured more)